src/command/verify/verify.c

```diff
@@ -0,0 +1,1124 @@
+
+                // Set current backup if there is one and verify the archive history on disk is in the database history
+                jobData.currentBackup = setBackupCheckArchive(
+                    jobData.backupList, backupInfo, jobData.archiveIdList, jobData.pgHistory, &jobData.jobErrorTotal);
+
+                // Create the parallel executor
+                ProtocolParallel *parallelExec = protocolParallelNew(
+                    (TimeMSec)(cfgOptionDbl(cfgOptProtocolTimeout) * MSEC_PER_SEC) / 2, verifyJobCallback, &jobData);
+
+                for (unsigned int processIdx = 1; processIdx <= cfgOptionUInt(cfgOptProcessMax); processIdx++)
+                    protocolParallelClientAdd(parallelExec, protocolLocalGet(protocolStorageTypeRepo, 1, processIdx));
+
+                // Process jobs
+                do
+                {
+                    unsigned int completed = protocolParallelProcess(parallelExec);
+
+                    // Process completed jobs
+                    for (unsigned int jobIdx = 0; jobIdx < completed; jobIdx++)
+                    {
+                        // Get the job and job key
+                        ProtocolParallelJob *job = protocolParallelResult(parallelExec);
+                        unsigned int processId = protocolParallelJobProcessId(job);
+                        StringList *filePathLst = strLstNewSplit(varStr(protocolParallelJobKey(job)), FSLASH_STR);
+                        // String *fileType = strLstGet(filePathLst, 0);
+                        strLstRemoveIdx(filePathLst, 0);
+                        String *filePathName = strLstJoin(filePathLst, "/");
+
+                        ArchiveResult *archiveIdResult = NULL;
+
+                        // Find the archiveId in the list - assert if not found since this should never happen
+                        String *archiveId = strLstGet(filePathLst, 0);
+                        unsigned int index = lstFindIdx(jobData.archiveIdResultList, &archiveId);
+                        ASSERT(index != LIST_NOT_FOUND);
+
+                        archiveIdResult = lstGet(jobData.archiveIdResultList, index);
+
+                        // The job was successful
+                        if (protocolParallelJobErrorCode(job) == 0)
+                        {
+                            const VerifyResult verifyResult = (VerifyResult)varUIntForce(protocolParallelJobResult(job));
+
+                            if (verifyResult == verifyOk)
+                                archiveIdResult->totalValidWal++;
+                            else
+                            {
+                                // Log a warning because the WAL may have gone missing if expire came through and removed it
+                                // legitimately so it is not necessarily an error so the jobErrorTotal should not be incremented
+                                if (verifyResult == verifyFileMissing)  // {uncovered - unable to cover in test harness}
```

Comment:
Yes, but part of my motivation was that I wanted it to be glaring how I was treating the results differently (missing - WARN, checksum/file/other - ERROR). So since you didn't mention that, I'll assume you're OK with it and yes, will move the IF block into a function to cover it.

Reply:
It think it will still be understandable in a function and we need the coverage for that warning.

Reply:
Ha - that wasn't exactly what I meant - and I certainly didn't type it out well enough now that I reread my comment here. I meant - are you OK with the missing file result being a warning but the filesize and checksum results being an error?

Reply:
I think that's OK for now. It might be better to figure out if we even care about that WAL file anymore when it goes missing (i.e. do any backups still depend on it), but I think that's out of scope for now.

Reply:
fixed 04f7263b738f396bf017f4f8be7ddf6f5cb09266